api/src/modules/document/listenerManager.ts

```diff
@@ -0,0 +1,71 @@
+import path from 'path';
+import fs from 'fs';
+import { DocumentDriveServer, InternalTransmitter } from 'document-drive';
+import {
+  Listener,
+  ListenerFilter,
+  actions,
+  reducer,
+  DocumentDriveDocument
+} from 'document-model-libs/document-drive';
+
+const listeners: Promise<any>[] = [];
+
+function loadModules(startPath: string, filter: string) {
+  if (!fs.existsSync(startPath)) {
+    console.log("no dir ", startPath);
+    return;
+  }
+
+  var files = fs.readdirSync(startPath);
+  for (var i = 0; i < files.length; i++) {
+    var filename = path.join(startPath, files[i]);
+    var stat = fs.lstatSync(filename);
+    if (stat.isDirectory()) {
+      loadModules(filename, filter); //recursive
+    } else if (filename.endsWith(filter)) {
+      listeners.push(import(filename));
+    };
+  };
+};
+
+function isListenerRegistered(drive: DocumentDriveDocument, listener: Listener) {
+  const listeners = drive.state.local.listeners;
+  return listeners.filter((l) => l.listenerId === listener.listenerId).length > 0;
+}
+
+async function registerListener(driveServer: DocumentDriveServer, driveId: string, listener: Listener) {
+  let drive = await driveServer.getDrive(driveId);
+
+  drive = reducer(drive, actions.addListener({ listener }));
```

Comment:
We need a better way to add operations. We also could have specific methods to add a listener ´addInternalListener(options, onTransmit)´